lib/pacemaker/pcmk_sched_resource.c

```diff
@@ -85,51 +85,52 @@ static resource_alloc_functions_t allocation_methods[] = {
     }
 };
 
-gboolean
-check_rsc_parameters(pe_resource_t * rsc, pe_node_t * node, xmlNode * rsc_entry,
-                     gboolean active_here, pe_working_set_t * data_set)
+/*!
+ * \internal
+ * \brief Check whether a resource's agent standard, provider, or type changed
+ *
+ * \param[in] rsc             Resource to check
+ * \param[in] node            Node needing unfencing/restart if agent changed
+ * \param[in] rsc_entry       XML with previously known agent information
+ * \param[in] active_on_node  Whether \p rsc is active on \p node
+ * \param[in] data_set
+ *
+ * \return true if agent for \p rsc changed, otherwise false
+ */
+bool
+pcmk__rsc_agent_changed(pe_resource_t *rsc, pe_node_t *node,
+                        const xmlNode *rsc_entry, bool active_on_node,
+                        pe_working_set_t *data_set)
```

Comment:
You could also drop the data_set parameter in favor of rsc->cluster here.

Reply:
added another commit on top for this